raiden/network/rpc/client.py

```diff
@@ -155,7 +156,12 @@ def format_data_for_call(
 def check_node_connection(func):
     """ A decorator to reconnect if the connection to the node is lost."""
     def retry_on_disconnect(self, *args, **kwargs):
+        """self here is always an obhect of JSONRPCClient"""
```

Comment:
This should be a comment instead of docstring.
`s/obhect/object/`

Reply:
I will change the typo, but don't agree on the docstring. self is part of the function description. Feel strange to write a comment below it explaining how the function's argument should behave.

Reply:
The docstring is for documentation, that is definitely /not/ adding any useful documentation because the behavior is defined by the decorated function, and not by the wrapper function.

If you want it to be meaningful to the caller you need to use the docstring from the decorated function, then the [functools.wrap](https://docs.python.org/2.7/library/functools.html#functools.wraps) is the right tool for the job

Reply:
BTW, if you want to tell the user the decorator should only be used to wrap a method of a `JSONRPCClient` class, then that needs to go in the `check_node_connection` function docstring, `retry_on_disconnect` is not part of a public interface.

Reply:
These functools.wrap sounds more complicated than it's worth.

Your 2nd comment sounds like the best way to put this one. Will put it in the decorator itself.